AnnoDesigner/Helper/PolygonBoundaryFinderHelper.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+
+namespace AnnoDesigner.Helper
+{
+    public static class PolygonBoundaryFinderHelper
+    {
+        private enum Direction
+        {
+            Up,
+            Left,
+            Down,
+            Right
+        }
+
+        private static Point GetLeftCell(Point point, Direction direction)
+        {
+            switch (direction)
+            {
+                case Direction.Up:
+                    return new Point(point.X - 1, point.Y - 1);
+                case Direction.Left:
+                    return new Point(point.X - 1, point.Y);
+                case Direction.Down:
+                    return point;
+                default:
+                    return new Point(point.X, point.Y - 1);
+            }
+        }
+
+        private static Point GetRightCell(Point point, Direction direction)
+        {
+            switch (direction)
+            {
+                case Direction.Up:
+                    return new Point(point.X, point.Y - 1);
+                case Direction.Left:
+                    return new Point(point.X - 1, point.Y - 1);
+                case Direction.Down:
+                    return new Point(point.X - 1, point.Y);
+                default:
+                    return point;
+            }
+        }
+
+        private static Point MoveForward(Point point, Direction direction)
+        {
+            switch (direction)
+            {
+                case Direction.Up:
+                    return new Point(point.X, point.Y - 1);
+                case Direction.Left:
+                    return new Point(point.X - 1, point.Y);
+                case Direction.Down:
+                    return new Point(point.X, point.Y + 1);
+                default:
+                    return new Point(point.X + 1, point.Y);
+            }
+        }
+
+        public static IList<Point> GetBoundaryPoints(ISet<Point> insidePoints)
+        {
+            var points = new List<Point>();
```

Comment:
I think it is better to name the variables according to their function. So maybe `var result =` would be better.
What do you think?